tests/integration/test_create_json_schema.py

```diff
@@ -0,0 +1,371 @@
+"""Tests for JSON Schema generation"""
+
+from typing import Generator, Any
+import json
+import uuid
+from time import sleep
+
+from synapseclient import Folder
+from synapseclient.client import Synapse
+import pytest
+
+from schematic.schemas.create_json_schema import (
+    create_json_schema,
+)
+
+from schematic.schemas.data_model_graph import DataModelGraph, DataModelGraphExplorer
+from schematic.schemas.data_model_parser import DataModelParser
+
+
+@pytest.fixture(name="schema_org", scope="module")
+def fixture_schema_org() -> Generator[str, None, None]:
+    """
+    This yields the Synapse org the test Schemas will be created at
+    """
+    yield "dpetest"
+
+
+@pytest.fixture(name="schema_version", scope="module")
+def fixture_schema_version() -> Generator[str, None, None]:
+    """
+    This the version to give all created test schemas
+    """
+    yield "0.0.1"
+
+
+@pytest.fixture(name="dmge", scope="module")
+def fixture_dmge() -> Generator[DataModelGraphExplorer, None, None]:
+    """
+    This yields DataModelGraphExplorer instance.
+    This has a module scope.
+    This allows for it to an input to other module-scoped fixtures.
+    """
+    data_model_parser = DataModelParser(
+        path_to_data_model="tests/data/example.model.csv"
+    )
+    parsed_data_model = data_model_parser.parse_model()
+    data_model_graph = DataModelGraph(
+        parsed_data_model, data_model_labels="class_label"
+    )
+    graph_data_model = data_model_graph.graph
+    return DataModelGraphExplorer(graph_data_model)
+
+
+@pytest.fixture(name="biospecimen_json_schema", scope="module")
+def fixture_biospecimen_json_schema(
+    synapse_module_scope: Synapse,
+    schema_org: str,
+    schema_version: str,
+    dmge: DataModelGraphExplorer,
+    request,
+):
+    """This creates a JSON Schema, uploads it to Synapse, then returns the uri"""
+    js = synapse_module_scope.service("json_schema")
+
+    schema = create_json_schema(
+        dmge=dmge,
+        datatype="Biospecimen",
+        schema_name="Biospecimen_validation",
+        write_schema=False,
+        use_property_display_names=False,
+    )
+    schema_name = upload_schema_to_synapse(
+        schema, synapse_module_scope, schema_org, schema_version
+    )
+
+    def delete_schema():
+        js.delete_json_schema(f"{schema_org}-{schema_name}")
+
+    request.addfinalizer(delete_schema)
```

Comment:
This `def delete_schema` is duplicated a bunch. If that could be extracted out and re-used it'll reduce some duplication, but it isn't a huge issue

Reply:
I'll try. I need to experiment with how finalizers work. Can I define them outside the fixture?

Reply:
I'm not sure tbh, and it isn't major - So I'd rather that you spend time working on other items and leave this